### PR TITLE
Add test coverage to token rec

### PR DESCRIPTION
**Pull request by efiacor**
Refactor existing test to use mockery helper functions
Add controller-runtime mock

### CODE REVIEW

File: controllers/pkg/reconcilers/token/reconciler.go [hunk 1]
```diff
@@ -45,7 +45,6 @@ func init() {
 const (
 	finalizer = "infra.nephio.org/finalizer"
 	// errors
-	errGetCr        = "cannot get cr"
```
Comment (liamfallon) [inline controllers/pkg/reconcilers/token/reconciler.go:48]: Grand, this is an unused constant.

File: controllers/pkg/.mockery.yaml [hunk 2]
```diff
@@ -1,6 +1,13 @@
+with-expecter: true
 packages:
   github.com/nephio-project/nephio/controllers/pkg/giteaclient:
     interfaces:
       GiteaClient:
         config:
-          dir: "{{.InterfaceDir}}"
\ No newline at end of file
+          dir: "{{.InterfaceDir}}"
+  sigs.k8s.io/controller-runtime/pkg/client:
```
Comment (liamfallon) [inline controllers/pkg/.mockery.yaml:8]: So you can mock interfaces outside your own packages as well, Nice!

**Comment by vjayaramrh** [comment]
@efiacor Are you planning to rebase and update the PR? Thanks

**Comment by dkosteck** [comment]
@efiacor General question: How would one go about testing this PR? Is the testing documented somewhere? I'm asking as a newcomer trying to get familiar with the process

**Comment by efiacor** [comment]
> @efiacor Are you planning to rebase and update the PR? Thanks

Done

**Comment by efiacor** [comment]
> @efiacor General question: How would one go about testing this PR? Is the testing documented somewhere? I'm asking as a newcomer trying to get familiar with the process

Hi @dkosteck , you can checkout the PR and simply run "make test" from the project root.
We ( @vjayaramrh @liamfallon ) started to introduce mockery to assist with mocking the api dependencies. 
@liamfallon added a nice guide here - https://github.com/nephio-project/docs/blob/main/content/en/docs/guides/contributor-guides/unit-testing-mockery.md on how we are using the mockery framework.

**Pull request merged**